pkg/scheduler/scheduling_partition.go

```diff
@@ -439,36 +440,45 @@ func (psc *partitionSchedulingContext) allocate(alloc *schedulingAllocation) boo
 // Called for both allocations from reserved as well as for direct allocations.
 // The unreserve is already handled before we get here so there is no difference in handling.
 // Lock free call this must be called holding the context lock
-func (psc *partitionSchedulingContext) confirmAllocation(appID, nodeID, allocKey string, confirm bool) error {
+func (psc *partitionSchedulingContext) confirmAllocation(proposal *commonevents.AllocationProposal, confirm bool) error {
```

Comment:
We also need to notify RMProxy about the confirmed allocation, previously Cache notifies RMProxy about confirmed allocation, now since scheduler will do confirmation and have chances to fail, we need to move the notification scheduler (once allocation is confirmed).

Reply:
It makes sense. I've moved the notification code to the scheduler, right after the proposal is confirmed.